tests/rados/test_slow_op_requests.py

```diff
@@ -1,67 +1,131 @@
+        )
 
-        pool = create_pools(config, rados_obj, client_node)
-        should_not_be_empty(pool, "Failed to retrieve pool details")
-        pools = config.get("create_pools")
-        for each_pool in pools:
-            cr_pool = each_pool["create_pool"]
-            osd_services = rados_obj.list_orch_services(service_type="osd")
-            for service in osd_services:
-                client_node.exec_command(cmd=f"ceph orch restart {service}", sudo=True)
-            for node in osd_nodes:
-                log.info(f"Rebooting node : {node.hostname}")
-                node.exec_command(sudo=True, cmd="reboot", check_ec=False)
-            method_should_succeed(rados_obj.bench_write, **cr_pool)
-        rados_obj.change_recovery_threads(config=pool, action="set")
+        slow_ops_frequency = {}
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=240)
+        while datetime.datetime.now() < endtime:
+            health_detail = rados_obj.run_ceph_command(
+                cmd="ceph health detail", client_exec=True
+            )
+            slow_ops_dict = health_detail.get("checks").get("SLOW_OPS")
+            if slow_ops_dict:
+                log.info("Slow Ops request found in ceph health detail")
+                log.info("Proceeding to update OSD list with frequency")
+                slow_ops_msg = slow_ops_dict.get("summary").get("message")
+            if slow_ops_msg:
+                daemon_str = re.search(r"daemons \[([^\]]+)\]", slow_ops_msg)
+                daemon_list = daemon_str.group(1).split(",")
+                log.info("Slow Ops daemons: ", daemon_list)
+                for _id in daemon_list:
+                    slow_ops_frequency[_id] = (
+                        slow_ops_frequency[_id] + 1
+                        if slow_ops_frequency.get(_id)
+                        else 1
+                    )
+            time.sleep(10)
 
         end_time, err = installer.exec_command(cmd="sudo date -u '+%Y-%m-%d %H:%M:%S'")
-        logs = get_slow_requests_log(installer, start_time.strip(), end_time.strip())
-        log.info(f"logs --- {logs}")
+        if not slow_ops_frequency:
+            log.error(
+                "Slow Ops warning was NOT generated/captured during the 240 secs smart wait"
+            )
+            raise Exception(
+                "Slow Ops warning was NOT generated/captured during the 120 secs smart wait"
+            )
+
+        log.info("Slow Ops request frequency: ", slow_ops_frequency)
+        max_osd = max(slow_ops_frequency)
+        log.info("OSD with highest count of Slow Ops: ", max_osd)
+
+        init_logs = rados_obj.get_journalctl_log(
+            start_time=start_time,
+            end_time=end_time,
+            daemon_type="osd",
+            daemon_id=max_osd.split(".")[-1],
+        )
+        slow_ops_log_list = []
+        for line in init_logs.splitlines():
+            if "slow ops" in line:
```

Comment:
I think the test should fail even if we do not find any occurrence of slow_ops on OSD logs. This is because we have already validated that the selected OSD had recorded most of the slow_ops